src/stan/lang/grammars/term_grammar_def.hpp

```diff
@@ -231,15 +231,17 @@ namespace stan {
         %= lexeme[char_("a-zA-Z")
                   >> *char_("a-zA-Z0-9_.")];
 
-      prob_args_r.name("probability function arguments");
+      prob_args_r.name("probability function argument");
       prob_args_r
         %= (lit('(') >> lit(')'))
         | hold[lit('(')
                >> expression_g(_r1)
                >> lit(')')]
         | (lit('(')
            >> expression_g(_r1)
-           >> lit('|')
+           >> (lit(',')
```

Comment:
Correct.  The `>>` is for sequencing and allows backtracking through it.  The `>` is for sequencing, but cuts (in the Prolog sense) backtracking.  The `require_vbar_f` just prints a message that you want the `|` instead of the `,` and then sets `_pass = false` so that the rule match fails.  The Qi non-terminal `eps` matches the empty sequence, so `eps > lit('|')` says that you just require the literal `|` and can't backtrack if you don't get it.